src/main/java/org/apache/sling/api/resource/ResourceUtil.java

```diff
@@ -271,18 +271,25 @@ public static String getParent(final String path, final int level) {
      * @throws NullPointerException If <code>path</code> is <code>null</code>.
      */
     public static @NotNull String getName(@NotNull String path) {
+        //        if (path == null) {
+        //            throw new NullPointerException("provided path is null");
+        //        }
```

Comment:
I think it makes sense to have this code in, it makes it more clear that a NPE is thrown and prevents that this contract breaks on refactoring

Reply:
Maybe `Objects.requireNonNull` though?